mlflow/gateway/handlers.py

```diff
@@ -0,0 +1,109 @@
+from enum import Enum
+from pathlib import Path
+from pydantic import BaseModel, Field, validator, parse_obj_as
+from pydantic.json import pydantic_encoder
+from typing import Optional, Union, List
+import yaml
+import json
+
+
+class Provider(str, Enum):
+    UNSPECIFIED_PROVIDER = "unspecified"
+    OPENAI = "openai"
+    ANTHROPIC = "anthropic"
+    BARD = "bard"
+
+
+class RouteType(str, Enum):
+    UNSPECIFIED = "unspecified"
+    LLM_V1_INSTRUCT = "llm/v1/instruct"
+    LLM_V1_CHAT = "llm/v1/chat"
+
+
+class OpenAIConfig(BaseModel):
+    openai_api_key: str = Field(..., alias="openai_api_key")
+    openai_api_type: Optional[str] = Field(None, alias="openai_api_type")
+    openai_api_base: Optional[str] = Field("https://api.openai.com/", alias="openai_api_base")
+    openai_api_version: Optional[str] = Field(None, alias="openai_api_version")
```

Comment:
Can we add `openai_organization`?

https://github.com/openai/openai-python/blob/77e5cf97933235b5567a5c9af0eac39f470b1f94/openai/__init__.py#LL48C32-L48C51

Reply:
good call-out. I'll make it optional